src/queryValidator.js

```diff
@@ -0,0 +1,67 @@
+import { keyTypes } from "./constants.js";
+import { createError } from "./error.js";
+
+export const queryValidator = (queryString) => {
+  const error = [];
+
+  // Check if the URL is encoded and if there is something before CMCD request
+  if (decodeURI(queryString) === queryString) {
+    error.push(createError("parameter-encoding"));
+  } else {
+    const query = queryString.split("?").pop();
+
```

Comment:
Let's rework the logic so instead of being like this:
```
if (...) {
error.push(...)
} else {
      ...
      if (....) {
      eror.push(...)
      } else {
      ......
      ....
      }
  return 
```
we do 
```
if (...)
return(error)
...
...
if(...)
return(error)
....
```
That way we avoid creating a big` if else if else` tree